templates/cloud_init/cloud_init.client.yml

```diff
@@ -0,0 +1,367 @@
+      chmod 755 ~/.terraform.d/plugins/registry.terraform.io/hashicorp/avere/$version/linux_amd64/terraform-provider-avere_v$version
+      [ $? -ne 0 ] && error "chmod2."
+
+      echo "mkdir tf"
+      mkdir tf
+      [ $? -ne 0 ] && error "mkdir tf."
+
+      echo "cd tf"
+      cd tf
+      [ $? -ne 0 ] && error "cd tf."
+
+      echo "git init"
+      git init
+      [ $? -ne 0 ] && error "git init."
+
+      echo "git remote"
+      git remote add origin -f https://github.com/Azure/Avere.git
+      [ $? -ne 0 ] && error "git remote."
+
+      echo "git config"
+      git config core.sparsecheckout true
+      [ $? -ne 0 ] && error "git config."
+
+      echo "src"
+      echo "src/terraform/*" >> .git/info/sparse-checkout
+      [ $? -ne 0 ] && error "src."
+
+      echo "git pull"
+      git pull origin main
+      [ $? -ne 0 ] && error "git pull."
+
+      echo "to file path"
+      cd src/terraform/examples/vfxt/${RUN_TERRAFORM_FILE}
+      [ $? -ne 0 ] && error "filepath."
+
+      echo "sskey"
+      sshkey="<ssh_key> azureuser@linuxvm"
+      [ $? -ne 0 ] && error "sshkey."
+
+      sed -i -E -e "s:vm_ssh_key_data = null:vm_ssh_key_data = \"${sshkey}\":" \
+        -e "s/location\s*= \"eastus\"/location = \"${REGION}\"/" \
+        -e "s/open_external_sources\s*= [\"*\"]/open_external_sources = [\"AzureCloud\"]/" \
+        -e "s/controller_add_public_ip\s*= true/controller_add_public_ip = false/" \
+        -e "s/controller_image_id\s*= null/controller_image_id = \"$CONTROLLER_IMAGE\"/" \
+        -e "s/vfxt_image_id\s*= null/vfxt_image_id = \"$VFXT_IMAGE\"/" \
+        -e "s/(([a-z]+)_resource_group_name)\s*= \"[^\"]+\"/\1 = \"${RESOURCE_GROUP_NAME}-\2\"/" \
+        -e "s/vfxt_cluster_password\s*= \"VFXT_PASSWORD\"/vfxt_cluster_password = \"${AVERE_ADMIN_PW}\"/" \
+        -e "s/peer_vnet_rg\s*= \"\"/peer_vnet_rg = \"$RESOURCE_GROUP_NAME\"/" \
+        -e "s/peer_vnet_name\s*= \"\"/peer_vnet_name = \"$VNET_NAME\"/" \
+        -e "s/storage_account_name\s*= \"storageaccount\"/storage_account_name = \"$RESOURCE_GROUP_NAME\"/" \
+        -e 's/vfxt_node_count\s*= 3/vfxt_node_count = <original_node_count>/' main.tf
+
+      terraform init
+      terraform apply -auto-approve
+
+  - path: /opt/scaleup.sh
+    permissions: '0755'
+    content: |
+      #!/bin/bash
+
+      function error() {
+        >&2 echo "echo >< ERROR: $* See log for details."
+        exit 1
+      }
+
+      export AZURE_SUBSCRIPTION='<sub_id>'
+      export ARM_SUBSCRIPTION_ID='<sub_id>'
+      export ARM_CLIENT_ID='<arm_client_id>'
+      export ARM_CLIENT_SECRET='<arm_client_scrt>'
+      export ARM_TENANT_ID='<arm_tenant_id>'
+      export RUN_TERRAFORM_FILE='<run_terraform_file>'
+
+      echo "az login"
+      az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID
+      [ $? -ne 0 ] && error "az login."
+
+      echo "az acct set"
+      az account set -s $AZURE_SUBSCRIPTION
+      [ $? -ne 0 ] && error "az acct set."
```

Comment:
Once the VM has logged in once (in setup_environment.sh), it shouldn't need to log in again every time.

_If removing this, it applies to other shell scripts in this cloud-init file._

Reply:
removed logins.